hazelcast/src/test/java/com/hazelcast/config/YamlConfigBuilderTest.java

```diff
@@ -556,6 +556,27 @@ public void readPortCount() {
         assertEquals(5703, config.getNetworkConfig().getPort());
     }
 
+
+    @Test(expected = SchemaViolationConfigurationException.class)
+    public void negativePort() {
+        Config config = buildConfig(""
+                + "hazelcast:\n"
+                + "  network:\n"
+                + "    port:\n"
+                + "      port-count: 200\n"
+                + "      port: -5701\n");
+    }
+
+    @Test(expected = SchemaViolationConfigurationException.class)
+    public void negativeCount() {
+        Config config = buildConfig(""
+                + "hazelcast:\n"
+                + "  network:\n"
+                + "    port:\n"
+                + "      port-count: -200\n"
+                + "      port: 5701\n");
+    }
```

Comment:
There is a comment at the top saying that

```
 * NOTE: This test class must not define test cases, it is meant only to
 * implement test cases defined in {@link AbstractConfigBuilderTest}.
```
So, I am not sure if it is OK to add new test cases here. 

Also, these are not really related to fix, so maybe we can move them to a separate PR. But, I am OK with having them here. Same for other new tests.

Reply:
I think have to  move tests to other pr

Reply:
What I meant is that, you should probably define the test cases in `AbstractConfigBuilderTest` and implement them here and in other necessary places. 

And yes, I believe it would be good to move them to another PR, as you move the test definition to the correct place, you would have to add XML config validation tests as well.

Reply:
ok. I removed tests from that pr. there is only NPE fix.
I am going to open another one with tests with XML and Yaml format.

thanks.